syntax = "proto3";

package kmounter;

// GENERAL

message ActionMessage {
    string message = 1;
}

// MOUNT

message Umount {
    string mount_name = 1;
    string home_sym_link = 2;
    bool force = 3;
}

message MountNFS {
    string mount_name = 1;
    string source = 2;
    string options = 3;
    string home_sym_link = 4;
}

// Generic mount meta-package
message Mount {
    string type = 1;
    oneof mount {
        MountNFS nfs = 2;
    }
}

// BACKUP

message BackupSync {
    Mount mount = 1;
    string target_to_sync = 2;
}

message StopSync {
    string target_to_sync = 1;
}
